lib/Runtime/ByteCode/ByteCodeEmitter.cpp

```diff
@@ -4875,6 +4880,11 @@ void ByteCodeGenerator::EmitPropStore(Js::RegSlot rhsLocation, Symbol *sym, Iden
     }
 
     // Arrived at the scope in which the property was defined.
+    if (sym && sym->GetNeedDeclaration() && scope->GetFunc() == funcInfo)
+    {
+        EmitUseBeforeDeclarationRuntimeError(this, Js::Constants::NoRegister);
+    }
```

Comment:
I'll leave a comment here to call out the intentional fall through

---
In reply to: [62564629](https://github.com/Microsoft/ChakraCore/pull/936#discussion_r62564629) [](ancestors = 62564629)